x-pack/plugins/ml/public/application/components/job_spaces_repair/job_spaces_repair_flyout.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useState, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiTitle,
+  EuiFlyoutBody,
+  EuiText,
+  EuiCallOut,
+  EuiSpacer,
+} from '@elastic/eui';
+
+import { ml } from '../../services/ml_api_service';
+import {
+  RepairSavedObjectResponse,
+  SavedObjectResult,
+} from '../../../../common/types/saved_objects';
+import { RepairList } from './repair_results';
+import { useToastNotificationService } from '../../services/toast_notification_service';
+
+interface Props {
+  onClose: () => void;
+}
+export const JobSpacesRepairFlyout: FC<Props> = ({ onClose }) => {
+  const { displayErrorToast, displaySuccessToast } = useToastNotificationService();
+  const [loading, setLoading] = useState(false);
+  const [repairable, setRepairable] = useState(false);
+  const [repairResp, setRepairResp] = useState<RepairSavedObjectResponse | null>(null);
+
+  async function loadRepairList(simulate: boolean = true) {
+    setLoading(true);
+    try {
+      const resp = await ml.savedObjects.repairSavedObjects(simulate);
+      setRepairResp(resp);
+
+      const count = Object.values(resp).reduce((acc, cur) => acc + Object.keys(cur).length, 0);
+      setRepairable(count > 0);
+      setLoading(false);
+      return resp;
+    } catch (error) {
+      // this shouldn't be hit as errors are returned per-repair task
+      // as part of the response
```

Comment:
Probably should set loading to false in the case of error as well so the user isn't left with a forever loading spinner.

Reply:
fixed in https://github.com/elastic/kibana/pull/83320/commits/b217713f58a6c1d4ef5189e7daf51592bd56693a